pkg/storage/stores/composite_store.go

```diff
@@ -171,6 +173,14 @@ func (c CompositeStore) GetChunks(ctx context.Context, userID string, from, thro
 		fetchers = append(fetchers, fetcher...)
 		return nil
 	})
+
+	// Protect ourselves against OOMing.
+	maxChunksPerQuery := c.limits.MaxChunksPerQueryFromStore(userID)
+	if maxChunksPerQuery > 0 && len(chunkIDs) > maxChunksPerQuery {
+		err := errors.QueryError(fmt.Sprintf("Query %v fetched too many chunks (%d > %d)", predicate, len(chunkIDs), maxChunksPerQuery))
+		return nil, nil, err
+	}
```

Comment:
I wonder if we could enforce this in the frontend. The index provides the number of chunks that will be queried.

Reply:
Yes, we could, but in this PR I just want to make `max_chunks_per_query` functional.

`max_chunks_per_query` is storage config, so I just add the code here to use store limit config. 

Can I move `max_chunks_per_query` config to querier config? If I could, and I move this code to querier validation function `validateQueryRequest(..)`.

Reply:
> max_chunks_per_query is storage config

then it's fine :slightly_smiling_face: